src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php

```diff
@@ -164,14 +164,19 @@ public function register_routes() {
 	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
 	 */
 	protected function get_parent( $parent_post_id ) {
+		$error = new WP_Error(
+			'rest_post_invalid_parent',
+			__( 'Invalid template parent ID.' ),
+			array( 'status' => 404 )
+		);
+
+		if ( (int) $parent_post_id <= 0 ) {
```

Comment:
I don't think we need the cast.

```suggestion
		if ( ! $parent_post_id ) {
```

Also, is it possible to add tests for this?

Reply:
The cast isn't something I came up with, it is a copy from the [WP_REST_Revisions_Controller](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L157). I thought there might be a reason why it is done that way? Do you still want me to change it?

I will look into adding test, but I have no experience writing unit tests for WP Core.

Reply:
Hmm okay then let's keep it for consistency.

There are actually already 9 test failures, see https://github.com/WordPress/wordpress-develop/actions/runs/8909919653/job/24468151188?pr=6472#step:14:381

For some of them it looks like the tests could be updated, but for others it could be that this actually causes a regression.

Existing REST API tests are a good point of reference. Also, consult https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/ and the #core-restapi channel on Slack if you need help

Reply:
I see, I made the incorrect assumption $parent_post_id would be an integer. Fixed that, which fixed the 9 test failures... And by the way it removed the cast ;)

I will look at unit tests tomorrow

Reply:
@swissspidy The unit test has been added, could you check if it is OK this way?